sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/code.py

```diff
@@ -2,12 +2,59 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
 
-from typing import Optional
+import os
+from typing import Dict, Optional, Union
+from pathlib import Path
 
+from azure.ai.ml._utils._asset_utils import get_content_hash, get_ignore_file, IgnoreFile
+from azure.ai.ml._utils._pathspec import GitWildMatchPattern
+from azure.ai.ml.constants._common import ArmConstants
 from ...entities._assets import Code
+from ...entities._component.code import ComponentCode, ComponentIgnoreFile
 
 
-class InternalCode(Code):
+class InternalComponentIgnoreFile(ComponentIgnoreFile):
+    _INTERNAL_COMPONENT_CODE_IGNORES = ["*.additional_includes"]
+
+    def __init__(self, file_path: Optional[Union[str, Path]] = None):
+        super(InternalComponentIgnoreFile, self).__init__(file_path=file_path)
+        self._path_spec.extend([GitWildMatchPattern(ignore) for ignore in self._INTERNAL_COMPONENT_CODE_IGNORES])
```

Comment:
so we still implement skipping __pycache__ via copy?

what about use this IgnoreFile in copy/compress directly?

Reply:
for additional includes, we do have copy/compress and it's OK to do so (and also easier to test)
but for standard v2 component, I think we cannot.